src/boot/pcrphase.c

```diff
@@ -175,21 +176,27 @@ static int run(int argc, char *argv[]) {
 
         length = strlen(word);
 
-        /* Skip logic if sd-stub is not used, after all PCR 11 might have a very different purpose then. */
-        r = efi_get_variable_string(EFI_LOADER_VARIABLE(StubPcrKernelImage), &pcr_string);
-        if (r == -ENOENT) {
-                log_info("Kernel stub did not measure kernel image into PCR %u, skipping measurement.", TPM_PCR_INDEX_KERNEL_IMAGE);
-                return EXIT_SUCCESS;
-        }
-        if (r < 0)
-                return log_error_errno(r, "Failed to read StubPcrKernelImage EFI variable: %m");
+        r = getenv_bool("SYSTEMD_PCRPHASE_FORCE");
+        if (r < 0 && r != -ENXIO)
+                log_warning_errno(r, "Unable to parse $SYSTEMD_PCRPHASE_FORCE value, ignoring.");
 
-        /* Let's validate that the stub announced PCR 11 as we expected. */
-        r = safe_atou(pcr_string, &pcr_nr);
-        if (r < 0)
-                return log_error_errno(r, "Failed to parse StubPcrKernelImage EFI variable: %s", pcr_string);
-        if (pcr_nr != TPM_PCR_INDEX_KERNEL_IMAGE)
-                return log_error_errno(SYNTHETIC_ERRNO(EREMOTE), "Kernel stub measured kernel image into PCR %u, which is different than expected %u.", pcr_nr, TPM_PCR_INDEX_KERNEL_IMAGE);
+        if (r <= 0) {
```

Comment:
Now if `SYSTEMD_PCRPHASE_FORCE=yes`, then the EFI variable never checked. If that's intended, I think it is better to rename `SYSTEMD_SKIP_STUB_PCR11` or so.

Or, unconditionally read the EFI variable, and when the environment variable is true, continue even when it does not exist:
```c
        r = efi_get_variable_string(EFI_LOADER_VARIABLE(StubPcrKernelImage), &pcr_string);
        if (r == -ENOENT) {
                log_info("Kernel stub did not measure kernel image into PCR %u, skipping measurement.", TPM_PCR_INDEX_KERNEL_IMAGE);
                r = getenv_bool("SYSTEMD_PCRPHASE_FORCE");
                if (r < 0 && r != -ENXIO)
                        log_warning_errno(r, "Unable to parse $SYSTEMD_PCRPHASE_FORCE value, ignoring.");
                if (r <= 0)
                        return EXIT_SUCCESS;
                goto tpm2_logic;
        }
        if (r < 0)
                return log_error_errno(r, "Failed to read StubPcrKernelImage EFI variable: %m");
```